ooDesign/frankensteinClass/solutionM/src/main/java/oointro/Frankenstein.java

```diff
@@ -0,0 +1,62 @@
+package oointro;
+
+public class Frankenstein {
+    private String cString;     // combination string of song and pet info
+    private double alphaAvg;    // average number of alphanumerics
+    private Song frankenSong;   // aggregate class
+    private Pet frankenPet;     // aggregate class
+
+
+    public void setSong(String song) {
+        // takes string representation of the song and parses it
+        String[] info = song.split(",");
+
+        frankenSong = new Song();
+
+        frankenSong.setTitle(info[0]);
+        frankenSong.setArtist(info[1]);
+        frankenSong.setReleaseYear(Integer.parseInt(info[2]));
+        frankenSong.setRating(Double.parseDouble(info[3]));
+    }
+
+    public String getSong() {
+        return frankenSong.toString();
+    }
+
+    public void setPet(String petName, String animalType, float age, String foodType) {
+
+        frankenPet = new Pet();
+
+        frankenPet.setName(petName);
+        frankenPet.setAnimalType(animalType);
+        frankenPet.setAge(age);
+        frankenPet.setFoodType(foodType);
+    }
+
+    public String getPet() {
+        return frankenPet.toString();
+    }
+
+    public void setString() {
+        cString = getSong() + "\n" + getPet();
+    }
+
+    public String getString() {
+        setString();
+
+        return cString;
+    }
+
+    public void setAlphaAvg() {
+        int totalAggStrings = 5;
+        String mergeStrings = frankenPet.getFoodType() + frankenPet.getName() + frankenPet.getAnimalType() + frankenSong.getTitle() + frankenSong.getArtist();
+
+        alphaAvg = mergeStrings.length() / totalAggStrings;
+    }
+
+    public double getAlphaAvg() {
+        setAlphaAvg();

```

Comment:
This is a clever way to solve the update problem!  I like it.    Don't change the code for this solution, but I'd be tempted to not have a setter at all and make a private helper method to calculateAlphaAverage instead.